imqsauth.go

```diff
@@ -234,6 +251,221 @@ func createDB(config *authaus.Config) bool {
+	errFlag := ""
+	if errPerms2 {
+		errFlag = "*"
+	}
+	fmt.Fprintf(w, "Nr of enabled users:\t%v%v\n", stats[enabled], errFlag)
+	fmt.Fprintf(w, "Nr of admin users:\t%v%v\n", stats[admin], errFlag)
+	fmt.Fprintf(w, "Nr of archived users:\t%v\n", stats[archived])
+
+	fmt.Fprintf(w, "User type counts (archived):\n")
+	typeKeys := maps.Keys(typeCounts)
+	slices.SortFunc(typeKeys, func(a authaus.AuthUserType, b authaus.AuthUserType) int { return int(a) - int(b) })
+	for _, v := range typeKeys {
+		fmt.Fprintf(w, "\tType %v(%v):\t%v\t(%v)\n", authaus.AuthUserTypeStrings[v], v, typeCounts[v].enabled, typeCounts[v].total)
+	}
+
+	// Error checks
+	// ------------
+
+	// Permits check
+	fmt.Fprintf(w, "Permits:\n")
+	plist, err := ic.Central.GetPermits()
+	if err != nil {
+		errorSet[err.Error()] = err
+		fmt.Fprintf(w, "\tPermit list ERROR\n")
+	} else {
+		fmt.Fprintf(w, "\tPermit list ok:\t%v permits\n", len(plist))
+		fmt.Fprintf(w, "\t*Nr of non-archived users without permits:\t%v\n", stats[unArchivedWithoutPermit])
+		fmt.Fprintf(w, "\t*Nr of archived users without permits:\t%v\n", len(users)-len(plist)-stats[unArchivedWithoutPermit])
+
+	}
+
+	// Non-existent groups
+	if len(missingGroups) > 0 {
+		fmt.Fprintf(w, "Missing groups:\n")
+	}
+
+	groupKeys := maps.Keys(missingGroups)
+	slices.SortFunc(groupKeys, func(a authaus.GroupIDU32, b authaus.GroupIDU32) int { return int(a) - int(b) })
+	for _, v := range groupKeys {
+		slices.SortFunc(missingGroups[v], func(a authaus.UserId, b authaus.UserId) int { return int(a) - int(b) })
+		fmt.Fprintf(w, "\tGroup missing:\t%v\tUsers: %v\n", v, missingGroups[v])
+	}
+
+	fmt.Fprintf(w, "\n* - indicates possibly inaccurate values.\n")
+
+	// Errors
+	fmt.Fprintln(w, "")
+	fmt.Fprintln(w, "Errors (explicit only - review the missing group list for all missing groups)")
+	fmt.Fprintln(w, "------")
+
+	errKeys := maps.Keys(errorSet)
+	sort.Strings(errKeys)
+
+	if len(errKeys) > 0 {
+		for _, e := range errKeys {
+			fmt.Fprintf(w, "%v\n", errorSet[e])
+		}
+	} else {
+		fmt.Fprintf(w, "No errors\n")
+	}
+
+	// FINAL HEALTHCHECK OUTPUT
+	w.Flush()
+	fmt.Print(sb.String())
+
+	// CLEAN MISSING GROUPS IF REQUESTED
+	if cleanAll {
+		for _, g := range groupKeys {
+			cleanGroup(ic, strconv.Itoa(int(g)))
+		}
+	}
+
+	return true
+}
+
+func getUserStats(ic *auth.ImqsCentral, users []authaus.AuthUser, stats map[string]int, typeCounts map[authaus.AuthUserType]*enabledTotal, errorSet map[string]error) (map[authaus.GroupIDU32][]authaus.UserId, bool, error) {
+	var err error
+	errPerms := false
+	missingGroups := map[authaus.GroupIDU32][]authaus.UserId{}
+	for _, u := range users {
+		p, _ := ic.Central.GetPermit(u.UserId)
+
+		if u.Archived {
+			stats[archived]++
+		}
+
+		if p == nil && !u.Archived {
+			stats[unArchivedWithoutPermit]++
+		}
+
+		typeCounts[u.Type].total++
+		if perm, ePerm := ic.Central.GetPermit(u.UserId); ePerm == nil {
+			// cycle through permit and check if group exists
+			if ePerm == nil {
```

Comment:
No need to check this again, as you only got to this point because `ePerm` was `nil`